src/latex.ts

```diff
@@ -105,6 +105,28 @@ function activateLatexPlugin(app: JupyterLab, manager: IDocumentManager, editorT
     let errorPanel: ErrorPanel | null = null;
     let pending = false;
 
+    const findOpenOrRevealPDF = () => {
+      let pdfWidget = manager.findWidget(pdfFilePath);
+      if (!pdfWidget) {
+        pdfWidget = manager.openOrReveal(pdfFilePath);
+      }
+      pdfContext = manager.contextForWidget(pdfWidget);
+      pdfContext.disposed.connect(cleanupPreviews);
+    };
+
+    const errorPanelInit = () => {
```

Comment:
This is nice! Some of the disposal logic here can be futher simplified when jupyterlab/jupyterlab#3499 lands.